### PR TITLE
journal: clean up the FSS related code a bit & add a couple of tests

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

File: src/libsystemd/sd-journal/journal-authenticate.c [hunk 1]
```diff
@@ -236,10 +248,9 @@ int journal_file_hmac_put_object(JournalFile *f, ObjectType type, Object *o, uin
                 r = journal_file_move_to_object(f, type, p, &o);
                 if (r < 0)
                         return r;
-        } else {
+        } else
```
Comment (bluca) [inline src/libsystemd/sd-journal/journal-authenticate.c]: if below can be merged in this line too

**Review by bluca — APPROVED** [review]

File: src/libsystemd/sd-journal/journal-authenticate.c (hunk 1 above)
Comment (mrc0mmand) [reply src/libsystemd/sd-journal/journal-authenticate.c]: Indeed!

File: src/libsystemd/sd-journal/journal-authenticate.c [hunk 2]
```diff
@@ -236,10 +248,8 @@ int journal_file_hmac_put_object(JournalFile *f, ObjectType type, Object *o, uin
                 r = journal_file_move_to_object(f, type, p, &o);
                 if (r < 0)
                         return r;
-        } else {
-                if (type > OBJECT_UNUSED && o->object.type != type)
+        } else if (type > OBJECT_UNUSED && o->object.type != type)
                         return -EBADMSG;
```
Comment (yuwata) [inline src/libsystemd/sd-journal/journal-authenticate.c]: indentation.

File: src/libsystemd/sd-journal/journal-authenticate.c [hunk 3]
```diff
@@ -13,6 +13,16 @@
 #include "memory-util.h"
 #include "time-util.h"
 
+static void* fssheader_free(FSSHeader *p) {
+        if (!p)
```
Comment (yuwata) [inline src/libsystemd/sd-journal/journal-authenticate.c]: How about `if (!p || p == MAP_FAILED)` ?

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/libsystemd/sd-journal/journal-authenticate.c (hunk 3 above)
Comment (mrc0mmand) [reply src/libsystemd/sd-journal/journal-authenticate.c]: Nice!

**Review by yuwata — APPROVED** [review]

**Pull request merged**